src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -248,26 +268,73 @@ def _maybe_warn_unowned(
     unowned_dependency_behavior: UnownedDependencyUsage,
     unowned_imports: Iterable[str],
     parsed_imports: ParsedPythonImports,
+    resolve: str,
+    project_uses_multiple_resolves: bool,
 ) -> None:
-    if unowned_imports and unowned_dependency_behavior is not UnownedDependencyUsage.DoNothing:
-        unowned_imports_with_lines = [
-            f"{module_name} ({file}:{parsed_imports[module_name].lineno})"
-            for module_name in sorted(unowned_imports)
-        ]
-        raise_error = unowned_dependency_behavior is UnownedDependencyUsage.RaiseError
-        log = logger.error if raise_error else logger.warning
-        log(
-            f"The following imports in {address} have no owners:\n\n{bullet_list(unowned_imports_with_lines)}\n\n"
-            "If you are expecting this import to be provided by your own firstparty code, ensure that it is contained within a source root. "
-            "Otherwise if you are using a requirements file, consider adding the relevant package.\n"
-            "Otherwise consider declaring a `python_requirement_library` target, which can then be inferred.\n"
-            f"See {doc_url('python-third-party-dependencies')}"
+    if not unowned_imports or unowned_dependency_behavior is UnownedDependencyUsage.DoNothing:
+        return
+    unowned_imports_with_lines = [
+        f"{module_name} (line: {parsed_imports[module_name].lineno})"
+        for module_name in sorted(unowned_imports)
+    ]
+    resolves_short_line = (
+        "\n  * Multiple resolves. See below." if project_uses_multiple_resolves else ""
+    )
+    resolves_long_line = (
+        (
+            "\n  * Multiple resolves. A target may exist for the import, but not share the same "
+            f"resolve of `{resolve}` from the target {address}. Pants only can infer dependencies "
+            f"on targets using the same resolve. You may need to set the `resolve=` field for "
+            f"whichever target provides that import (and possibly use the `parametrize` mechanism "
+            f"if that code should work with multiple resolves). See "
+            f"{doc_url('python-third-party-dependencies#multiple-lockfiles')}."
         )
-
-        if raise_error:
-            raise UnownedDependencyError(
-                "One or more unowned dependencies detected. Check logs for more details."
-            )
+        if project_uses_multiple_resolves
+        else ""
+    )
+    msg = softwrap(
```

Comment:
As briefly discussed in Slack, this is a long message....NB that most users will have to scroll up in their terminal to read the start. An alternative could be linking to the docsite and having a short description here, but I figured a Python specific guide is useful and this helps people on planes.

I'm concerned how chatty things are if set to `warn` rather than `error` because you might have several of these.